ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -160,8 +170,10 @@ void onComplete(DeleteOperation op) {
     }
     routerMetrics.operationDequeuingRate.mark();
     routerMetrics.deleteBlobOperationLatencyMs.update(time.milliseconds() - op.getSubmissionTimeMs());
-    operationCompleteCallback
-        .completeOperation(op.getFutureResult(), op.getCallback(), op.getOperationResult(), op.getOperationException());
+    if (op.setCallbackInvoked()) {
```

Comment:
rather than doing it this way (here as well as in the `PutManager`), do you think it is better to call `op.invokeOperationCompleteCallback()` which does the check and callback invocation internally? Having a `setCallbackInvoked()` looks a bit odd.